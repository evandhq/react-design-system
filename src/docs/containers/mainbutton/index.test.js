import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import MainButtonPage from './index';

describe('snapshot testing for all examples on doc page', () => {
  it('mainButton', () => {
    const wrapper = render(<MainButtonPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
