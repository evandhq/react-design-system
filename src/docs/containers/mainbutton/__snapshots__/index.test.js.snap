// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing for all examples on doc page mainButton 1`] = `
Array [
  <h1
    class="sc-bdVaJa irmrqS"
    data-test="h1-tag"
  >
    دکمه ها
  </h1>,
  <br />,
  <br />,
  <h2
    class="sc-bwzfXH hlvhiK"
    data-test="h2-tag"
  >
    نمونه دکمه پیشفرض
  </h2>,
  <button
    class=" button-primary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه
    </span>
  </button>,
  <br />,
  <h2
    class="sc-bwzfXH hlvhiK"
    data-test="h2-tag"
  >
    دکمه آیکون دار
  </h2>,
  <button
    class=" button-primary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه آیکون دار
    </span>
  </button>,
  <br />,
  <h2
    class="sc-bwzfXH hlvhiK"
    data-test="h2-tag"
  >
    دکمه full-width
  </h2>,
  <button
    class=" button-primary sc-chPdSV hzkfVM"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs jbxVbw"
      color="white"
      data-test="span"
      size="10"
    >
      دکمه full-width (sm size)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV kZfPpu"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه full-width(md size - default)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV jlObrS"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs doVAst"
      color="white"
      data-test="span"
      size="20"
    >
      دکمه full-width(lg size)
    </span>
  </button>,
  <br />,
  <h2
    class="sc-bwzfXH hlvhiK"
    data-test="h2-tag"
  >
    دکمه در اندازه های متفاوت
  </h2>,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    sm size
  </h3>,
  <button
    class=" button-primary sc-chPdSV krqVKK"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs jbxVbw"
      color="white"
      data-test="span"
      size="10"
    >
      دکمه
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV krqVKK"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY dtLGbe"
    >
      <i
        class="icon-watch_later sc-cSHVUG guKhFg"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs jbxVbw"
      color="white"
      data-test="span"
      size="10"
    >
      دکمه
    </span>
  </button>,
  <br />,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    md size (default)
  </h3>,
  <button
    class=" button-primary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه
    </span>
  </button>,
  <br />,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    lg size
  </h3>,
  <button
    class=" button-primary sc-chPdSV kmaCYc"
    data-test="button"
    type="submit"
  >
    <span
      class="sc-htoDjs doVAst"
      color="white"
      data-test="span"
      size="20"
    >
      دکمه
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV kmaCYc"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY ciMFBd"
    >
      <i
        class="icon-watch_later sc-cSHVUG dadeXC"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs doVAst"
      color="white"
      data-test="span"
      size="20"
    >
      دکمه
    </span>
  </button>,
  <br />,
  <h2
    class="sc-bwzfXH hlvhiK"
    data-test="h2-tag"
  >
    دکمه ها در استایل ها متفاوت و رنگ های متفاوت
  </h2>,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    رنگ قرمز
  </h3>,
  <button
    class=" button-primary sc-chPdSV jlraTZ"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV bFLryg"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default) - isDisabled
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV jlraTZ"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG hNuDoP"
        color="red"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs dJtNTe"
      color="red"
      data-test="span"
      size="14"
    >
      دکمه secondary
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV bFLryg"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG hNuDoP"
        color="red"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs dJtNTe"
      color="red"
      data-test="span"
      size="14"
    >
      دکمه secondary - isDisabled
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV jlraTZ"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV bFLryg"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary - isDisabled
    </span>
  </button>,
  <hr />,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    رنگ سبز
  </h3>,
  <button
    class=" button-primary sc-chPdSV jRAzAm"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV eNuQdL"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default) - isDisabled
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV jRAzAm"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG laYqjl"
        color="green"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs fcTmNH"
      color="green"
      data-test="span"
      size="14"
    >
      دکمه secondary
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV eNuQdL"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG laYqjl"
        color="green"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs fcTmNH"
      color="green"
      data-test="span"
      size="14"
    >
      دکمه secondary - isDisabled
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV jRAzAm"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV eNuQdL"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary - isDisabled
    </span>
  </button>,
  <hr />,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    رنگ زرد
  </h3>,
  <button
    class=" button-primary sc-chPdSV KbShx"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV gSpYdg"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default) - isDisabled
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV KbShx"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG fEAIhM"
        color="yellow"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs ijahRT"
      color="yellow"
      data-test="span"
      size="14"
    >
      دکمه secondary
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV gSpYdg"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG fEAIhM"
        color="yellow"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs ijahRT"
      color="yellow"
      data-test="span"
      size="14"
    >
      دکمه secondary - isDisabled
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV KbShx"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV gSpYdg"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary - isDisabled
    </span>
  </button>,
  <hr />,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    رنگ آبی
  </h3>,
  <button
    class=" button-primary sc-chPdSV kGfofH"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV hctkVs"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default) - isDisabled
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV kGfofH"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG jXgbPU"
        color="blue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs ghkjrx"
      color="blue"
      data-test="span"
      size="14"
    >
      دکمه secondary
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV hctkVs"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG jXgbPU"
        color="blue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs ghkjrx"
      color="blue"
      data-test="span"
      size="14"
    >
      دکمه secondary - isDisabled
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV kGfofH"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV hctkVs"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary - isDisabled
    </span>
  </button>,
  <hr />,
  <h3
    class="sc-htpNat djFCwm"
    data-test="h3-tag"
  >
    رنگ آبی تیره - پیشفرض
  </h3>,
  <button
    class=" button-primary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default)
    </span>
  </button>,
  <button
    class=" button-primary sc-chPdSV hAhUzu"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG cqpyEq"
        color="white"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs cemEaf"
      color="white"
      data-test="span"
      size="14"
    >
      دکمه primary (default) - isDisabled
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه secondary
    </span>
  </button>,
  <button
    class=" button-secondary sc-chPdSV hAhUzu"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه secondary - isDisabled
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV iOIAtU"
    data-test="button"
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary
    </span>
  </button>,
  <button
    class=" button-tertiary sc-chPdSV hAhUzu"
    data-test="button"
    disabled=""
    type="submit"
  >
    <div
      class="sc-kgoBCf dLBqYr sc-kAzzGY gnTUdE"
    >
      <i
        class="icon-watch_later sc-cSHVUG dYkkKn"
        color="darkBlue"
        data-test="icon"
      />
    </div>
    <span
      class="sc-htoDjs deFOOE"
      color="darkBlue"
      data-test="span"
      size="14"
    >
      دکمه tertiary - isDisabled
    </span>
  </button>,
  <br />,
]
`;
